_search-plugins/neural-sparse-search.md

```diff
@@ -0,0 +1,208 @@
+
+When selecting a model, choose one of the following options:
+
+- Use a sparse encoding model at both ingestion time and search time (high performance, relatively high latency).
+- Use a sparse encoding model at ingestion time and a tokenizer model at search time (low performance, relatively low latency).
+
+**PREREQUISITE**<br>
+Before using sparse search, you must set up a sparse embedding model. For more information, see [Using custom models within OpenSearch]({{site.url}}{{site.baseurl}}/ml-commons-plugin/ml-framework/).
+{: .note}
+
+## Using sparse search
+
+To use sparse search, follow these steps:
+
+1. [Create an ingest pipeline](#step-1-create-an-ingest-pipeline).
+1. [Create an index for ingestion](#step-2-create-an-index-for-ingestion).
+1. [Ingest documents into the index](#step-3-ingest-documents-into-the-index).
+1. [Search the index using neural search](#step-4-search-the-index-using-neural-search).
+
+## Step 1: Create an ingest pipeline
+
+To generate vector embeddings, you need to create an [ingest pipeline]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/) that contains a [`sparse_encoding` processor]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/processors/sparse-encoding/), which will convert the text in a document field to vector embeddings. The processor's `field_map` determines the input fields from which to generate vector embeddings and the output fields in which to store the embeddings.
+
+The following example request creates an ingest pipeline where the text from `passage_text` will be converted into text embeddings and the embeddings will be stored in `passage_embedding`:
+
+```json
+PUT /_ingest/pipeline/nlp-ingest-pipeline-sparse
+{
+  "description": "An sparse encoding ingest pipeline",
+  "processors": [
+    {
+      "sparse_encoding": {
+        "model_id": "aP2Q8ooBpBj3wT4HVS8a",
+        "field_map": {
+          "passage_text": "passage_embedding"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+## Step 2: Create an index for ingestion
+
+In order to use the text embedding processor defined in your pipelines, create a rank features index, adding the pipeline created in the previous step as the default pipeline. Ensure that the fields defined in the `field_map` are mapped as correct types. Continuing with the example, the `passage_embedding` field must be mapped as a k-NN vector with a dimension that matches the model dimension. Similarly, the `passage_text` field should be mapped as `text`.
```

Comment:
the `passage_embedding` field must be mapped as a k-NN vector with a dimension that matches the model dimension -> the `passage_embedding` field must be mapped as a rank_features type.